cmd/eksctl-anywhere/cmd/upgradepackages.go

```diff
@@ -46,20 +46,5 @@ func upgradePackages(ctx context.Context) error {
 		return fmt.Errorf("unable to initialize executables: %v", err)
 	}
 
-	b := curatedpackages.NewBundleReader(
-		kubeConfig,
-		ipo.source,
-		deps.Kubectl,
-		nil,
-		nil,
-	)
-	activeController, err := b.GetActiveController(ctx)
-	if err != nil {
-		return err
-	}
-	err = b.UpgradeBundle(ctx, activeController, upo.bundleVersion)
-	if err != nil {
-		return err
-	}
-	return nil
+	return curatedpackages.UpgradePackages(ctx, kubeConfig, deps.Kubectl, ipo.source, upo.bundleVersion)
```

Comment:
I matched the input from the code that was already there. Can you elaborate on what the bug is?

Reply:
In the suggestion I made, you can see that I've changed "ipo" to "upo". That's basically the bug. The variable "ipo", is a package-scoped variable in the "curatedpackages" package. It's name, "ipo", stands for "install package options". It's a struct containing a number of configuration options used when installing packages (obviously, I guess :)). Anyway, it has a member, "source". Meanwhile, there's another package-scoped variable in the same package, named "upo". It stands for "upgrade package options". I imagine you can guess what it's purpose is. It too contains a member named "source". However, it's value for source is different from "ipo"'s value for source. Since they're both package-scoped variables that are structs containing members named source, the compiler can't know that you want the one versus the other, so it doesn't even know to warn you. All of our CLI commands use this kind of package-scope variable for options and the command definitions, and they really should be refactored not to, for this very reason.

Since this line of code (line #49) is calling "UpgradePackages", it should be using the upgrade package options, ie "upo.source", as opposed to the install package options source, which would be "ipo.source". If for no other reason, than because "ipo.source" is probably a default or zero-value, whereas "upo.source" should contain whatever value was indicated by the user on the command-line.

Reply:
As it is today, `upgradePackageOptions` only has one field (`bundleVersion`) and no `source` so I can't apply the suggestiong as it won't compile. Re-structuring the CLI options is something I'd consider out-of-scope for this PR.